frontend/src/logger/pcm.logger.ts

```diff
@@ -0,0 +1,75 @@
+import {AxiosInstance} from 'axios'
+import {AppConfig} from '../app-config/types'
+import {handleNotAuthorizedErrors} from '../auth/handleNotAuthorizedErrors'
+import identityFn from 'lodash/identity'
+
+enum LogLevel {
+  info = 'INFO',
+  warning = 'WARNING',
+  debug = 'DEBUG',
+  error = 'ERROR',
+  critical = 'CRITICAL',
+}
+
+interface LogEntry {
+  message: string
+  level: LogLevel
+  extra?: {[key: string]: string}
+}
+
+export class PcmLogger {
+  private readonly post
+  private readonly handle401and403
+
+  constructor(axios: AxiosInstance, appConfig?: AppConfig) {
+    this.post = axios.post
+    this.handle401and403 = appConfig
+      ? handleNotAuthorizedErrors(appConfig)
+      : identityFn<Promise<any>>
+  }
+
+  log(logLevel: LogLevel, message: string, extra?: {[key: string]: string}) {
+    const logEntry = this.buildMessage(logLevel, message, extra)
+    return this.handle401and403(
+      this.post('/logs', logEntry, {
+        headers: {'Content-Type': 'application/json'},
+      }),
```

Comment:
Sadly this logic is repeated in our `request` function. I guess you did not use the `request` function here since it also adds the region to the URL. I would consider to avoid spreading configuration logic and error handling around. Can be done in another PR though